nearcore/src/cold_storage.rs

```diff
@@ -224,65 +224,88 @@ enum ColdStoreInitialMigrationResult {
     MigrationInterrupted,
 }
 
-/// This function performs initial population of cold storage if needed.
+/// This function performs migration to cold storage if needed.
 /// Migration can be interrupted via `keep_going` flag.
 ///
-/// First, checks that hot store is of kind `Archive`. If not, no migration needed.
-/// Then, captures hot final head BEFORE the migration, as migration is performed during normal neard run.
-/// If hot final head is not set, returns Err.
-/// Otherwise:
-/// 1. performed migration
-/// 2. updates head to saved hot final head
+/// Migration is performed if cold storage does not have a head set.
+/// New head is determined based on hot storage DBKind.
+/// - If hot storage is of type `Archive`, we need to perform initial migration from legacy archival node.
+///   This process will take a long time. Cold head will be set to hot final head BEFORE the migration started.
```

Comment:
Why do we need to do this? It's unintuitive to set the cold head before the migration. How does it play with interruptions?

Reply:
Ah looking at the code I see now what is going on. We determine the target cold head for the migration to copy everything from the genesis up to the target cold head and set it to the hot head in this case. But we actually write to to the db only after the migration is finished. Can you update the comment to reflect this?